modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/http/p2p/middlewares/MetricsMiddleware.scala

```diff
@@ -0,0 +1,104 @@
+package io.constellationnetwork.node.shared.http.p2p.middlewares
+
+import cats.data.Kleisli
+import cats.effect.kernel.Async
```

Comment:
using the `cats.effect.kernel.Async` sometimes causes compilation issues (at least in intellij). Can we switch this to `cats.effect.Async` instead?